webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -427,16 +426,25 @@ const AdvancedTextEditor = ({
         };
     }, [handleDraftChange, draft]);
 
-    // Set the draft from store when changing post or channels, and store the previus one
+    // Keep track of the draft as a ref so that we can save it when changing channels
+    const draftRef = useRef(draftFromStore);
     useEffect(() => {
-        setDraft(draftFromStore);
-        return () => handleDraftChange(previousDraft.current, {instant: true, show: true});
-    }, [channelId, postId]);
+        draftRef.current = draft;
+    }, [draft]);
 
-    // Keep track of the previous draft
+    // Set the draft from store when changing post or channels, and store the previous one
     useEffect(() => {
-        previousDraft.current = draft;
-    }, [draft]);
+        // Store the draft that existed when we opened the channel to know if it should be saved
+        const draftOnOpen = draftFromStore;
```

Comment:
This assignment is unneeded (unless you are doing it for pure readability reasons). The `draftFromStore` is always going to be the same as `draftOnOpen` in this context, so there is no need to assign it to another variable.

Reply:
Yeah, it's for readability. I kept expecting `draft` to contain the latest value from the state even though I understand why it's not, so I thought it would be good to be as clear as possible that that isn't the case